mautic/mautic
# Setup codeception and add e2e tests for contacts

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description
This PR introduces the setup for an end-to-end (E2E) test suite using DDEV, along with tests for contact management(create, view, delete, change segments, batch operations etc). We are using codeception for e2e test suite.







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Setup the project using ddev.
3. `ddev start`
4. `ddev ssh`
5. `bin/codecept run acceptance ContactManagementCest -vvv`
6. Ensure that all tests pass, confirming the creation and viewing of contacts work correctly.
